src/stdlib_str2num.f90

```diff
@@ -0,0 +1,532 @@
+!> The `stdlib_str2num` module provides procedures and interfaces for conversion
+!> of characters to numerical types. Currently supported: int32, real32 and real64
+!>
+!> This code was modified from https://github.com/jalvesz/Fortran-String-to-Num by Alves Jose
+!> And was possible thanks to all the discussions in this thread https://fortran-lang.discourse.group/t/faster-string-to-double/
+!>
+!> Known precisions limits of current proposal :
+!> Conversion to double precision is exact up to epsilon(0.0_dp)
+!>   example:
+!>   input          : 123456.78901234567890123456789012345678901234567890+2
+!>   formatted read : 12345678.90123457
+!>   to_num         : 12345678.90123457
+!>   difference abs :                 0.1862645149230957E-08
+!>   difference rel :                 0.1508742584455759E-13%
+!>    
+!> Conversion to quadruple precision can deviate at about 200*epsilon(0.0_qp)
+!>   example:
+!>   input          : 0.140129846432481707092372958328991613128026194187651577175706828388979108268586060148663818836212158203125E-443
+!>   formatted read : 0.140129846432481707092372958328991608E-443
+!>   to_num         : 0.140129846432481707092372958328996233E-443
+!>   difference abs :                                 0.4625E-475 
+!>   difference rel :                                 0.3300E-029%
+
+module stdlib_str2num
+    use iso_fortran_env, only: sp => real32, dp => real64, qp => real128
```

Comment:
Would it be possible to use the ones defined in `stdlib_kinds` instead of redefining them here (mainly for portability)?

Reply:
Yes sure, I used this because even though I launched the preprocessing for gfortran and intel I saw that qp was defined as -1 when doing my tests, so I used iso_fortran_env to ensure having it defined. This can be changed, But I guess the best option would be that the qp code is not written if the selected compiler does not support qp. Could you guide me on how to do that in this scope with fypp?

Reply:
I open a [PR](https://github.com/jalvesz/stdlib/pull/1) in your project with a suggestion with `fypp`. don't hesitate to ask questions there or here.